src/_guides/openapi/specification/v3.1/the-perfect-modern-openapi-workflow.md

```diff
@@ -0,0 +1,660 @@
+API instance of your choosing. This could be production if you are brave, or
+some other staging/testing environment, but the logic is simple:
+
+1. Send HTTP requests.
+2. See if that fails unexpectedly. 
+3. Receive HTTP response.
+4. See if that matches the OpenAPI document.
+
+This could be automated to run at regular intervals, or it could be triggered to
+run on pull requests and merges to make sure that any and all API or OpenAPI
+changes agree with each other. GitHub Actions is once again a good way to get
+this done.
+
+```yaml
+# .github/workflows/contract-testing.yml
+name: API Contract Testing
+on: [push]
+jobs:
+  contract-testing:
+    name: Deploy API documentation on Bump.sh
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - uses: microcks/import-github-action@v1
+        with:
+          specificationFiles: 'api/openapi.yaml:true'
+          microcksURL: 'https://mocks.example.com/api/'
+          keycloakClientId:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT }}
+          keycloakClientSecret:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT_CREDENTIALS }}
+
+      - uses: microcks/test-github-action@v1
+        with:
+          apiNameAndVersion: 'Train Travel API:1.0.0'
+          testEndpoint: 'http://api-testing.example.com'
+          runner: OPEN_API_SCHEMA
+          microcksURL: 'https://mocks.example.com/api/'
+          keycloakClientId:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT }}
+          keycloakClientSecret:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT_CREDENTIALS }}
+          waitFor: '10sec'
+```
+
+This can replace the `.github/workflows/api-mocks.yaml` workflow above which
+only wanted to update the mocks, allowing you to update moth mocks and run the
+contract testing to make sure the real API matches the OpenAPI that describes
+it, which also happens to ensure that the mock server is matching the actual API
+too.
+
+*Learn more about [contract testing with Microcks](_guides/bump-sh-tutorials/testing-with-microcks.md).*
+
+### Deploying Artifacts
+
+There are countless artifacts that need to be kept up-to-date with the API as it
+is changed over time, and instead of doing any of this manually we can let
+OpenAPI handle all of it.
+
+#### Deploy Documentation
+
+Using OpenAPI is the easiest way to maintain up-to-date documentation, without
+having to remember to make manual changes to some wiki/CMS somewhere, or try to
+time updates with code deployments. Tools like Bump.sh provide hosted API
+documentation which can be updated every time the Git repo receives updated
+OpenAPI.
+
+As soon as a pull request is merged to the `main` branch, the OpenAPI document
+that accompanies the source code can be deployed to the Bump.sh documentation,
+keeping everything in sync.
+
+```yaml
+# .github/workflows/api-docs.yml
+name: Deploy API documentation
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy-openapi:
+    if: ${{ github.event_name == 'push' }}
```

Comment:
this condition shouldn't be needed in this workflow file because the file only listens to `push` events on the `main` branch (`on.push.branches[].main`).